### ISSUE TITLE
Using SQL Azure for persistence fails for V6 with NRE

### ISSUE THREAD

**Issue by ramonsmits** [Particular/NServiceBus.NHibernate#201]
If using SQL Azure this fails with the following NullReferenceException with NServiceBus V6. Not an issue with V5.

It looks like it is related to promoting to a distributed transaction. The transport used is MSMQ. 
# Exception info

```
2016-06-08 13:13:50.9464|INFO|12|NServiceBus.FirstLevelRetriesBehavior|First Level Retry is going to retry message '58fcada3-b163-4f12-a4ca-a61f00fafc62' because of an exception:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Transactions.Transaction.GetPromotedToken()
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Data.SqlClient.SqlInternalConnection.EnlistNonNull(Transaction tx)
   at System.Data.SqlClient.SqlInternalConnection.Enlist(Transaction tx)
   at System.Data.ProviderBase.DbConnectionInternal.ActivateConnection(Transaction transaction)
   at System.Data.ProviderBase.DbConnectionPool.PrepareConnection(DbConnection owningObject, DbConnectionInternal obj, Transaction transaction)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
   at System.Data.SqlClient.SqlConnection.Open()
   at NHibernate.Connection.DriverConnectionProvider.GetConnection()
   at NServiceBus.Persistence.NHibernate.NHibernateSynchronizedStorageAdapter.TryAdapt(TransportTransaction transportTransaction, ContextBag context)
   at NServiceBus.LoadHandlersConnector.<AdaptOrOpenNewSynchronizedStorageSession>d__2.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\Incoming\LoadHandlersConnector.cs:line 56
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.LoadHandlersConnector.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\Incoming\LoadHandlersConnector.cs:line 26
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.InvokeSagaNotFoundBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\Sagas\InvokeSagaNotFoundBehavior.cs:line 16
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.MutateIncomingMessageBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\MutateInstanceMessage\MutateIncomingMessageBehavior.cs:line 28
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.DeserializeLogicalMessagesConnector.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\Incoming\DeserializeLogicalMessagesConnector.cs:line 30
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.SubscriptionReceiverBehavior.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Routing\MessageDrivenSubscriptions\SubscriptionReceiverBehavior.cs:line 30
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.MutateIncomingTransportMessageBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\MutateTransportMessage\MutateIncomingTransportMessageBehavior.cs:line 27
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.UnitOfWorkBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\UnitOfWork\UnitOfWorkBehavior.cs:line 26
--- End of stack trace from previous location where exception was thrown ---
   at NServiceBus.UnitOfWorkBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\UnitOfWork\UnitOfWorkBehavior.cs:line 48
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.ProcessingStatisticsBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\Performance\Statistics\ProcessingStatisticsBehavior.cs:line 25
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.CriticalTimeBehavior.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Performance\Statistics\CriticalTime\CriticalTimeBehavior.cs:line 16
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.ReceivePerformanceDiagnosticsBehavior.<Invoke>d__2.MoveNext() in C:\Build\src\NServiceBus.Core\Performance\Statistics\ReceivePerformanceDiagnosticsBehavior.cs:line 40
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.TransportReceiveToPhysicalMessageProcessingConnector.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\Incoming\TransportReceiveToPhysicalMessageProcessingConnector.cs:line 38
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at NServiceBus.Performance.StatisticsBehavior.<Invoke>d__2.MoveNext() in S:\particular\EndToEnd\src\PerformanceTests\Common\Statistics\StatisticsBehavior.V6.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.FirstLevelRetriesBehavior.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Recoverability\FirstLevelRetries\FirstLevelRetriesBehavior.cs:line 21
```

**Comment by DavidBoike** [comment]
@Particular/nhibernate-persistence-maintainers Is this still a concern for V6 RC?

**Comment by SzymonPobiega** [comment]
Not sure. I guess it needs a repro on the current version.

**Comment by ramonsmits** [comment]
Still occuring with core beta 7 and nhibernate beta 6.

**Comment by brunobertechini** [comment]
I am not sure if my issue is the same problem here. Please point me out if I need to open a new Issue.

* Transport: SQL Server (using SQL Azure)
* Persistence: NHibernate (using NHibernate4.SqlAzure driver) [https://docs.particular.net/nservicebus/nhibernate/sql-azure](url)
* Web site running on Azure AppService

Scenario:

1. Web sends a command
2. Endpoint Handler process command and publish an event
3. A Saga should be started with this event (IAmStartedByMessages). This saga has a custom Saga finder using Synchronized Session:
```
public async Task<MySagaData> FindBy(MyEvent message, SynchronizedStorageSession storageSession, ReadOnlyContextBag context)
        {
            var sagaContext = context as IInvokeHandlerContext;
            int tenantId = Convert.ToInt32(sagaContext.MessageHeaders[MessageHeaders.TenantId]);

            var dbSession = storageSession.Session();
            return await GetSagaFromDB(dbSession, tenantId, message.ConnectionId);
        }

private async Task<MySagaData> GetSagaFromDB(ISession session, int tenantId, Guid connectionId)
        {
            var saga = session.QueryOver<SkyHubNotSyncedOrderImporterSagaData>()
                              .Where(x => x.TenantId == tenantId && x.ConnectionId == connectionId)
                              .SingleOrDefault();

            // If a saga can't be found Task.FromResult(null) should be returned
            return await Task.FromResult(saga);
        }


```

And this fails with exception:

```
2016-11-20T13:14:47  PID[4344] Information System.NotSupportedException: Specified method is not supported.
2016-11-20T13:14:47  PID[4344] Information    at System.Data.Common.DbConnection.EnlistTransaction(Transaction transaction)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.Persistence.NHibernate.NHibernateSynchronizedStorageAdapter.OpenConnection(SessionFactoryImpl sessionFactoryImpl, Transaction ambientTransaction) in C:\Build\src\NServiceBus.NHibernate\SynchronizedStorage\NHibernateSynchronizedStorageAdapter.cs:line 55
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.Persistence.NHibernate.NHibernateSynchronizedStorageAdapter.<>c__DisplayClass4_0.<TryAdapt>b__0() in C:\Build\src\NServiceBus.NHibernate\SynchronizedStorage\NHibernateSynchronizedStorageAdapter.cs:line 49
2016-11-20T13:14:47  PID[4344] Information    at System.Lazy`1.CreateValue()
2016-11-20T13:14:47  PID[4344] Information    at System.Lazy`1.LazyInitValue()
2016-11-20T13:14:47  PID[4344] Information    at System.Lazy`1.get_Value()
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.Persistence.NHibernate.NHibernateLazyAmbientTransactionSynchronizedStorageSession.<>c__DisplayClass2_0.<.ctor>b__0() in C:\Build\src\NServiceBus.NHibernate\SynchronizedStorage\NHibernateAmbientTransactionSynchronizedStorageSession.cs:line 19
2016-11-20T13:14:47  PID[4344] Information    at System.Lazy`1.CreateValue()
2016-11-20T13:14:47  PID[4344] Information    at System.Lazy`1.LazyInitValue()
2016-11-20T13:14:47  PID[4344] Information    at System.Lazy`1.get_Value()
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.Persistence.NHibernate.NHibernateLazyAmbientTransactionSynchronizedStorageSession.get_Session() in C:\Build\src\NServiceBus.NHibernate\SynchronizedStorage\NHibernateAmbientTransactionSynchronizedStorageSession.cs:line 22
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.SynchronizedStorageSessionExtensions.Session(SynchronizedStorageSession session) in C:\Build\src\NServiceBus.NHibernate\SynchronizedStorageSessionExtensions.cs:line 20
2016-11-20T13:14:47  PID[4344] Information    at MySagaFinder.<FindBy>d__0.MoveNext() in ****\MySagaFinder.cs:line 27 (Removed sensitive data)
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.CustomFinderAdapter`2.<Find>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\Sagas\CustomFinderAdapter.cs:line 0
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.SagaPersistenceBehavior.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Sagas\SagaPersistenceBehavior.cs:line 42
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.LoadHandlersConnector.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\Incoming\LoadHandlersConnector.cs:line 40
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.InvokeSagaNotFoundBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\Sagas\InvokeSagaNotFoundBehavior.cs:line 16
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.MutateIncomingMessageBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\MutateInstanceMessage\MutateIncomingMessageBehavior.cs:line 28
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.DeserializeLogicalMessagesConnector.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\Incoming\DeserializeLogicalMessagesConnector.cs:line 31
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.InvokeAuditPipelineBehavior.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Audit\InvokeAuditPipelineBehavior.cs:line 18
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.SubscriptionReceiverBehavior.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Routing\MessageDrivenSubscriptions\SubscriptionReceiverBehavior.cs:line 29
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.MutateIncomingTransportMessageBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\MutateTransportMessage\MutateIncomingTransportMessageBehavior.cs:line 27
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.UnitOfWorkBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\UnitOfWork\UnitOfWorkBehavior.cs:line 26
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.UnitOfWorkBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\UnitOfWork\UnitOfWorkBehavior.cs:line 48
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.ProcessingStatisticsBehavior.<Invoke>d__0.MoveNext() in C:\Build\src\NServiceBus.Core\Performance\Statistics\ProcessingStatisticsBehavior.cs:line 25
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.ReceivePerformanceDiagnosticsBehavior.<Invoke>d__2.MoveNext() in C:\Build\src\NServiceBus.Core\Performance\Statistics\ReceivePerformanceDiagnosticsBehavior.cs:line 40
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at Abp.NServiceBus.AbpNServiceBusUnitOfWork.<Invoke>d__1.MoveNext() in C:\a\1\s\libs\Abp.NServiceBus\Abp.NServiceBus\UnitOfWork\AbpNServiceBusUnitOfWork.cs:line 49
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.TransportReceiveToPhysicalMessageProcessingConnector.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\Incoming\TransportReceiveToPhysicalMessageProcessingConnector.cs:line 37
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.MainPipelineExecutor.<Invoke>d__1.MoveNext() in C:\Build\src\NServiceBus.Core\Pipeline\MainPipelineExecutor.cs:line 32
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.Transport.SQLServer.ReceiveStrategy.<TryProcessingMessage>d__12.MoveNext() in C:\Build\src\NServiceBus.SqlServer\Receiving\ReceiveStrategy.cs:line 33
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2016-11-20T13:14:47  PID[4344] Information    at NServiceBus.Transport.SQLServer.ReceiveWithTransactionScope.<TryProcess>d__4.MoveNext() in C:\Build\src\NServiceBus.SqlServer\Receiving\ReceiveWithTransactionScope.cs:line 99
2016-11-20T13:14:47  PID[4344] Information --- End of stack trace from previous location where exception was thrown ---
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2016-11-20T13:14:47  PID[4344] Information    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

**Comment by SzymonPobiega** [comment]
@brunobertechini The exception you see is probably due to the fact SQL Azure does not support TransctionScope. Can you try setting you endpoint transaction mode to `sends-atomic-with-receive` (https://docs.particular.net/nservicebus/transports/transactions#transactions-transport-transaction-sends-atomic-with-receive) or enabling the outbox?

Please note that in `sends-atomic-with-receive` you will have to manage possible duplicate messages yourself to ensure correct behavior. In the outbox mode NServiceBus will manage deduplication for you.

**Comment by MarcinHoppe** [comment]
@SzymonPobiega See #244. Let's move the discussion there.

**Comment by SzymonPobiega** [comment]
@MarcinHoppe since we closed #244 should we also close this one?

**Comment by MarcinHoppe** [comment]
I think so. We have not seen any evidence that this is still a problem.
